### PR TITLE
[TypeScript][WebSocket] Update routerDialog

### PR DESCRIPTION
**Pull request by Ezecriis**
### Purpose
*What is the context of this pull request? Why is it being done?*

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

#### Bots
- [ ] I have validated that new and updated responses use appropriate [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) and [InputHint](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) properties to ensure a high-quality speech-first experience
- [ ] I have replicated language model changes across the English, French, Italian, German, Spanish, and Chinese `.lu` files and validated that deployment is successful

#### Deployment Scripts
- [ ] I have replicated my changes in the **Virtual Assistant Template** and **Sample** projects
- [ ] I have replicated my changes in the **Skill Template** and **Sample** projects

### CODE REVIEW

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialog.ts [hunk 1]
```diff
@@ -37,31 +39,34 @@ export abstract class RouterDialog extends InterruptableDialog {
...
                                 break;
                             }
                             case DialogTurnStatus.complete: {
-                                await this.complete(innerDc, result);
-                                // End active dialog
-                                await innerDc.endDialog();
-                                break;
+                                // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/tslint/config
+                                const dialogTurnResult: RouterDialogTurnResult = result.result;
+                                if (typeof dialogTurnResult === typeof RouterDialogTurnResult
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialog.ts]: I don't think that this is the correct way to check is `dialogTurnResult` is a type of `RouterDialogTurnResult`.

Here you have an examples of C# and TS:
>https://github.com/microsoft/botframework-solutions/blob/master/lib/csharp/microsoft.bot.builder.solutions/microsoft.bot.builder.solutions/Authentication/MultiProviderAuthDialog.cs#L285

> https://github.com/microsoft/botframework-solutions/blob/master/lib/typescript/botbuilder-solutions/src/authentication/multiProviderAuthDialog.ts#L262

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialog.ts [hunk 2]
```diff
@@ -37,31 +39,34 @@ export abstract class RouterDialog extends InterruptableDialog {
...
+                                    await this.route(innerDc);
+                                    break;
+                                }
                             }
                             default:
                         }
                     }
                     break;
                 }
+
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialog.ts]: This should be deleted

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialog.ts [hunk 3]
```diff
@@ -74,6 +79,12 @@ export abstract class RouterDialog extends InterruptableDialog {
                 default: {
                     await this.onSystemMessage(innerDc);
                 }
+
+            }
+            // If the active dialog was ended on this turn (either on single-turn dialog, or on continueDialogAsync)
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialog.ts]: This should not be here

Looks how the C# is managing this
> https://github.com/microsoft/botframework-solutions/blame/next/sdk/csharp/libraries/microsoft.bot.builder.solutions/Dialogs/RouterDialog.cs#L87

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnResult.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Copyright(c) Microsoft Corporation.All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { RouterDialogTurnStatus } from './routerDialogTurnStatus';
+
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnResult.ts]: Have you missed to comment the behavior of this class?

> https://github.com/microsoft/botframework-solutions/blob/master/lib/csharp/microsoft.bot.builder.solutions/microsoft.bot.builder.solutions/Dialogs/RouterDialogTurnResult.cs#L4

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnResult.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+export class RouterDialogTurnResult {
+
+    public constructor (status: RouterDialogTurnStatus) {
+        this.status = status;
+
+    }
+    public status: RouterDialogTurnStatus;
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnResult.ts]: As convention, the properties should be before the constructor

File: sdk/typescript/libraries/botbuilder-solutions/src/fallbackRequestProvider.ts [hunk 6]
```diff
@@ -11,4 +11,4 @@ import { TurnContext } from 'botbuilder';
 
 export interface IFallbackRequestProvider {
     sendRemoteFallbackEvent(turnContext: TurnContext): Promise<void>;
-}
\ No newline at end of file
+}
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/fallbackRequestProvider.ts]: This is okay.. but we should not update this file in this PR, maybe after the migration we will add a task to fix the eslint issues

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnStatus.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright(c) Microsoft Corporation.All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/// <summary>
+/// Define router dialog turn status.
+/// </summary>
+
+export enum RouterDialogTurnStatus {
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnStatus.ts]: We don't use `enum`, as replace you should use `namespace` like here 
> https://github.com/microsoft/botframework-solutions/blob/master/lib/typescript/botbuilder-skills/src/models/skillEvents.ts#L6

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnStatus.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright(c) Microsoft Corporation.All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/// <summary>
+/// Define router dialog turn status.
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnStatus.ts]: The `<summary>` is for C#.

Here you have an example of TypeScript one

> https://github.com/microsoft/botframework-solutions/blob/master/lib/typescript/botbuilder-solutions/src/dialogs/interruptionAction.ts#L8

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnStatus.ts [hunk 9]
```diff
@@ -3,9 +3,9 @@
...
  */
 
-/// <summary>
-/// Define router dialog turn status.
-/// </summary>
+/*
+* Define router dialog turn status.
+*/
 
 export enum RouterDialogTurnStatus {
```
Comment (VictorGrycuk) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnStatus.ts]: Change `enum` for `namespace`

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialog.ts [hunk 10]
```diff
@@ -37,29 +38,35 @@ export abstract class RouterDialog extends InterruptableDialog {
...
                                 break;
                             }
                             case DialogTurnStatus.complete: {
-                                await this.complete(innerDc, result);
-                                // End active dialog
-                                await innerDc.endDialog();
-                                break;
+                                // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/tslint/config
+                                const routerDialogTurnResult: RouterDialogTurnResult = <RouterDialogTurnResult> result.result;
+                                if (routerDialogTurnResult !== undefined) {
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialog.ts]: Do you miss the second condition of the `if` statement?

Please follow 
> https://github.com/microsoft/botframework-solutions/blame/next/sdk/csharp/libraries/microsoft.bot.builder.solutions/Dialogs/RouterDialog.cs#L69

Comment (Batta32) [reply sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialog.ts]:
```suggestion
                                if (routerDialogTurnResult !== undefined && routerDialogTurnResult.status === RouterDialogTurnStatus.Restart) {
```

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialog.ts [hunk 11]
```diff
@@ -74,6 +81,7 @@ export abstract class RouterDialog extends InterruptableDialog {
                 default: {
                     await this.onSystemMessage(innerDc);
                 }
+
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialog.ts]: This should be deleted

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnResult.ts [hunk 12]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright(c) Microsoft Corporation.All rights reserved.
+ * Licensed under the MIT License.
+ */
+/**
+ * Define router dialog turn result.
+ */
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnResult.ts]: The comments should be above of the `export class..`

File: sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnStatus.ts [hunk 13]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright(c) Microsoft Corporation.All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/*
+* Define router dialog turn status.
+*/
+
```
Comment (Batta32) [inline sdk/typescript/libraries/botbuilder-solutions/src/dialogs/routerDialogTurnStatus.ts]: This line should be deleted

**Comment by VictorGrycuk** [comment]
`routerDialogTurnResult` and `routerDialogTurnStatus` are not being exported in the index file

**Comment by Batta32** [comment]
LGTM!

**Pull request merged**